main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -456,12 +456,20 @@ object ConstraintGen {
       case e: Expr.RestrictableTag => RestrictableChooseConstraintGen.visitApplyRestrictableTag(e)
 
       case KindedAst.Expr.ExtensibleTag(label, exps, tvar, loc) =>
-        ??? // TODO: Ext-Variants
+        val pred = Name.Pred(label.name, label.loc)
+        val (tpes, effs) = exps.map(visitExp).unzip
+        val rest = Type.freshVar(Kind.SchemaRow, loc)
+        val row = Type.mkSchemaRowExtend(pred, Type.mkTuplish(tpes, loc), rest, loc)
```

Comment:
In terms of the backend - This should probably not be a maybe tuple, because then you cannot differentiate between a singleton enum with a tuple and an n-ary tuple post monomorphization

Reply:
Does this mean I get my Christmas wish of predicates becoming n-ary?

Reply:
Sorry, I am confused. What do you propose?

Reply:
> Does this mean I get my Christmas wish of predicates becoming n-ary?

We really should